### PR TITLE
audit contexts

### PR DESCRIPTION
**Pull request by jmank88**
https://smartcontract-it.atlassian.net/browse/BCF-3204

Supports:
- https://github.com/smartcontractkit/chainlink/pull/14920

### CODE REVIEW

File: pkg/v3/flows/conditional_test.go
Comment (jmank88) [inline pkg/v3/flows/conditional_test.go:1]: The mock constructors automatically register the `*testing.T` and `AssertExpectations` on `t.Cleanup`.

File: pkg/v3/types/mocks/metadatastore.generated.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Code generated by mockery v2.28.1. DO NOT EDIT.
+// Code generated by mockery v2.43.2. DO NOT EDIT.
```
Comment (jmank88) [inline pkg/v3/types/mocks/metadatastore.generated.go:1]: I didn't mean to use a different version here. Shouldn't CI be failing?

Comment (jmank88) [reply pkg/v3/types/mocks/metadatastore.generated.go:1]: `.tool-versions` was on 2.12.1 even though these files were on 2.28.1

Comment (jmank88) [reply pkg/v3/types/mocks/metadatastore.generated.go:1]: Bumped to 2.43.2

File: pkg/v3/tickers/time.go [hunk 2]
```diff
@@ -38,13 +40,13 @@ func NewTimeTicker[T any](interval time.Duration, observer observer[T], getterFn
...
-
-	if !t.closer.Store(cancel) {
-		return fmt.Errorf("already running")
+func (t *timeTicker[T]) Start(ctx context.Context) error {
+	if err := t.StartOnce("timeTicker", func() error { return nil }); err != nil {
+		return err
 	}
+	defer close(t.done)
+	ctx, cancel := t.stopCh.Ctx(ctx)
+	defer cancel()
```
Comment (infiloop2) [inline pkg/v3/tickers/time.go:49]: can't comment on the exact lines, 
in lines 64-66 (in the for loop) should there be a similar statement as above file 
```
if ctx.Err() != nil {
 					return nil
 				}
```

**Review by infiloop2 — COMMENTED** [review]
looks good, added a small comment

File: pkg/v3/tickers/time.go (hunk 2 above)
Comment (jmank88) [reply pkg/v3/tickers/time.go:49]: Updated to `continue`

Comment (infiloop2) [reply pkg/v3/tickers/time.go:49]: yeah that does seem like a bug. thanks for fixing!

**Review by infiloop2 — DISMISSED** [review]

File: pkg/v3/tickers/time.go (hunk 2 above)
Comment (jmank88) [reply pkg/v3/tickers/time.go:49]: Had to fix a test too

**Review by infiloop2 — APPROVED** [review]

**Pull request merged**